snuba/query/processors/sampling_rate.py

```diff
@@ -0,0 +1,28 @@
+from snuba import settings
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.request.request_settings import RequestSettings, SamplingMode
+
+
+class SamplingRateProcessor(QueryProcessor):
+    """
+    Applies the appropriate sampling rate to the query to be executed.
+    This will be a storage specific processor as soon as we will have a clickhouse
+    specific query class and ast. As long as that does not exist, this
+    is just applied at the very end of the query processing phase.
+
+    Right now it does not much, but it will be the place where the adaptive sampling
+    will be calculated and applied.
+    """
+
+    def process_query(self, query: Query, request_settings: RequestSettings,) -> None:
+        if not query.get_data_source().supports_sample():
```

Comment:
I feel like the decision to support or not support sampling belongs in the dataset in most cases. Like 2 datasets that share a source can independently choose to support or not supoprt sampling. I agree that the actual sample rate is by source - especially for Discover.

Reply:
correct.